sed/binning.py

```diff
@@ -355,7 +355,85 @@ def _hist_from_bin_ranges(
             j = (sample[t, i] - ranges[i, 0]) * delta[i]
             is_inside = is_inside and (0 <= j < bins[i])
             flatidx += int(j) * strides[i]
+            if not is_inside:
```

Comment:
I'd put comments before the statements. In this case, before the `if not is_inside`